### PR TITLE
Rework ticks computation for log scales

### PR DESCRIPTION
**Pull request by t-bltg**
Required for https://github.com/JuliaPlots/Plots.jl/pull/3577.

### CODE REVIEW

File: src/ticks.jl [hunk 1]
```diff
@@ -201,8 +207,8 @@ function optimize_ticks_typed(x_min::T, x_max::T, extend_ticks,
                         end
                         # round only those values that end up as viewmin and viewmax
                         # to save computation time
-                        S[k + 1] = round(S[k + 1], sigdigits = sigdigits(z))
-                        S[2 * k] = round(S[2 * k], sigdigits = sigdigits(z))
+                        S[k + 1] = round(S[k + 1], sigdigits=sigdigits(z), base=Int(base))
```
Comment (daschw) [inline src/ticks.jl:210]: Does this not fail for `base=ℯ`?

Comment (t-bltg) [reply src/ticks.jl:210]: It does, good catch ! Rounding using the `base` keyword requires an Integer. So what I propose is to use the base kw for `log2` and `log10 `and omit the kw for `log` ?

**Comment by BeastyBlacksmith** [comment]
why did you convert to draft?

**Comment by t-bltg** [comment]
> why did you convert to draft?

Some minor fixes on route, I'm on it,  don't merge it yet.

**Comment by t-bltg** [comment]
Ready :rocket:

**Pull request merged**